portal/models/questionnaire_bank.py

```diff
@@ -638,6 +685,23 @@ def add_if_not_found(self, commit_immediately=False):
         return self
 
 
+@dogpile_cache.region('qb_query_cache')
```

Comment:
Are you caching DB objects? I know we've had issues with stale DB objects with celery

Reply:
yes - and i believe i found the workaround - not the updated comment and all clients refresh the cached objects in the session.

Reply:
I am finding the *win* by caching these is minimal (like 15% over the entire run) - tempted to remove this caching as it does add complications.

Reply:
I *did* remove this caching - premature optimization, more trouble to get right than the minor savings, which we only see on the *first* run.